### PR TITLE
Providing templates and documentation on how to use the secure-foward…

### PR DESCRIPTION
**Pull request by ewolinetz**
… plugin

### CODE REVIEW

**Comment by ewolinetz** [comment]
@richm @sosiouxme @jcantrill PTAL

[test]

File: hack/testing/test-fluentd-forward.sh [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+  wait_for_pod_ACTION stop $fpod
+
+  # edit so we don't send to ES
+  oc get configmap/logging-fluentd -o yaml | sed '/## matches/ a \
+      < match ** >\
+        @include ../user/secure-forward.conf\
+      </match>' | oc replace -f -
+
+  # update configmap secure-forward.conf
+  oc patch configmap/logging-fluentd --type=json --patch '[{ "op": "replace", "path": "/data/secure-forward.conf", "value": "\
```
Comment (ewolinetz) [inline hack/testing/test-fluentd-forward.sh]: this should probably just be the `cat` of the test/template... //FIXME

**Comment by ewolinetz** [comment]
fixed build issue, let's [test]

**Comment by ewolinetz** [comment]
new test wasn't run since it lacked 'x' permissions on the file

File: deployer/README.md [hunk 2]
```diff
@@ -492,6 +494,34 @@ To set the parameters:
...
 
+#### Secure Forward
+
+To be able to accommodate different possible configurations for aggregated logging
+stacks, yet still maintain a light container we are now providing the [secure forward](https://github.com/tagomoris/fluent-plugin-secure-forward)
+plugin to enable customers to forward Fluentd logs from OpenShift's Fluentd to
+an external aggregator which can interpret Fluentd's messages and can then further
+process logs and/or forward them to a log store.
+
+To set up `secure forward`, you need only update the `secure-forward.conf` section of
```
Comment (jcantrill) [inline deployer/README.md]: "To set up `secure forward`, update the `secure-forward.conf` section of the config map provided...."  Not sure where its provided from....

In general, I would suggest writing less with using 'you' as its just the configuration and not necessarily this person's per se.

File: deployer/README.md [hunk 3]
```diff
@@ -492,6 +494,34 @@ To set the parameters:
...
+the config map provided for you. A sample, commented out, configuration has been
+provided. Further explanation of how to set up `secure forward` can be found
+at it's [repo](https://github.com/tagomoris/fluent-plugin-secure-forward).
```
Comment (jcantrill) [inline deployer/README.md]: at [repo]...

File: deployer/README.md [hunk 4]
```diff
@@ -492,6 +494,34 @@ To set the parameters:
...
+
+  $ oc edit configmap/logging-fluentd
+
+To add certificates to your Fluentd pods to be used in `secure forward` you can
```
Comment (jcantrill) [inline deployer/README.md]: Certificates can be used with 'secure forward' and Fluentd by updating the secret that is mounted into the pod.

File: deployer/README.md [hunk 5]
```diff
@@ -492,6 +494,34 @@ To set the parameters:
...
+update the secret that is mounted to the Fluentd pods.
+
+Avoid using secret names such as 'cert', 'key', 'ca' so not to clobber the keys
+that were generated by the Deployer pod for Fluentd to talk to the OSE hosted
+Elasticsearch.
+
+  $ oc patch secrets/logging-fluentd --type=json --patch "[{'op':'add','path':'/data/your_ca_cert','value':'$(base64 /path/to/your_ca_cert.pem)'}]"
+  $ oc patch secrets/logging-fluentd --type=json --patch "[{'op':'add','path':'/data/your_private_key','value':'$(base64 /path/to/your_private_key.pem)'}]"
+
+Note that the name provided on your path above (e.g. your_ca_cert, your_private_key)
```
Comment (jcantrill) [inline deployer/README.md]: ...the name provided on the path above....

File: hack/testing/test-fluentd-forward.sh [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    fi
+    return 0
+}
+
+get_running_pod() {
+    # $1 is component for selector
+    oc get pods -l component=$1 | awk -v sel=$1 '$1 ~ sel && $3 == "Running" {print $1}'
+}
+
+wait_for_pod_ACTION() {
```
Comment (jcantrill) [inline hack/testing/test-fluentd-forward.sh]: Intentional the last part it is capitalized?

Comment (richm) [inline hack/testing/test-fluentd-forward.sh]: @jcantrill no, just a copy/paste from somewhere else where it was ALL CAPS

Comment (ewolinetz) [inline hack/testing/test-fluentd-forward.sh]: I'll make it lowercase

**Comment by ewolinetz** [comment]
fluentd failed to stop in time during out cleanup... changed how we restart slightly

**Comment by ewolinetz** [comment]
comments addressed and tests are (running and) passing.

**Comment by ewolinetz** [comment]
[merge]

**Pull request merged**